src/utils/experimentUpdatesHandler.js

```diff
@@ -1,23 +1,48 @@
-import { updateProcessingSettings, updatePipelineStatus } from '../redux/actions/experimentSettings';
+import { updateProcessingSettings, updateBackendStatus } from '../redux/actions/experimentSettings';
 import loadEmbedding from '../redux/actions/embedding/loadEmbedding';
 import updatePlotData from '../redux/actions/componentConfig/updatePlotData';
 
+const updateTypes = {
+  DATA_PROCESSING: 'dataProcessing',
+  GEM2S: 'gem2s',
+};
+
 const experimentUpdatesHandler = (dispatch) => (experimentId, update) => {
-  const { input, output, status } = update;
+  dispatch(updateBackendStatus(experimentId, update.status));
+
+  switch (update.type) {
+    case updateTypes.DATA_PROCESSING: {
+      return onDataProcessingUpdate(experimentId, update, dispatch);
+    }
+    case updateTypes.GEM2S: {
+      return onGEM2SUpdate(experimentId, update, dispatch);
+    }
+    default: {
+      console.log(`Error, unrecognized message type ${update.type}`);
+    }
+  }
+};
+
+const onDataProcessingUpdate = (experimentId, update, dispatch) => {
+  const { input, output } = update;
 
   const processingConfigUpdate = output.config;
+  if (processingConfigUpdate) {
+    dispatch(updateProcessingSettings(experimentId, input.taskName, processingConfigUpdate));
+
+    Object.entries(output.plotData).forEach(([plotUuid, plotData]) => {
+      dispatch(updatePlotData(plotUuid, plotData));
+    });
 
-  dispatch(updatePipelineStatus(experimentId, status));
-  dispatch(updateProcessingSettings(experimentId, input.taskName, processingConfigUpdate));
+    // This is temporary, should be taken out eventually as it will be handled
+    // once we clean up the redux store (which is the correct action to take here)
+    dispatch(loadEmbedding(experimentId, 'umap'));
+    dispatch(loadEmbedding(experimentId, 'tsne'));
```

Comment:
Do you know why this should be dispatched? Is it still necessary? Or can we remove it?

Reply:
I think we had a conversation about this some weeks ago, wasn't it the thing that for some reason when removed the pipeline broke down?

Reply:
Oh right!